src/telegram_fmt_plug.c

```diff
@@ -164,44 +97,84 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 {
 	const int count = *pcount;
 	int index;
+	const int inc = (cur_salt->version == 1) ? SHA1_LOOP_CNT : SHA512_LOOP_CNT;
 
 	if (any_cracked) {
 		memset(cracked, 0, cracked_size);
 		any_cracked = 0;
 	}
+
 #ifdef _OPENMP
 #pragma omp parallel for
 #endif
-	for (index = 0; index < count; index += MIN_KEYS_PER_CRYPT) {
+	for (index = 0; index < count; index += inc) {
 		unsigned char pkey[MIN_KEYS_PER_CRYPT][256]; /* 2048 bits, yes */
```

Comment:
Looks like we only need 136, not 256, right? Perhaps move this to a `#define` and use that both here and in `pbkdf2_*()` calls.